tests/test_derivatives.py

```diff
@@ -1,44 +1,65 @@
 import numpy as np
 import pytest
-from conftest import t, x, y, z, skipif_yask
+from conftest import skipif_yask
 from sympy import Derivative, simplify
 
 from devito import Grid, Function, TimeFunction
 
 
 @pytest.fixture
-def shape(xdim=20, ydim=30):
-    return (xdim, ydim)
+def shape(xdim=20, ydim=20, zdim=20):
+    return (xdim, ydim, zdim)
 
 
 @pytest.fixture
 def grid(shape):
     return Grid(shape=shape)
 
 
+@pytest.fixture
+def x(grid):
+    return grid.dimensions[0]
+
+
+@pytest.fixture
+def y(grid):
+    return grid.dimensions[1]
+
+
+@pytest.fixture
+def z(grid):
+    return grid.dimensions[2]
+
+
+@pytest.fixture
+def t(grid):
+    return grid.stepping_dim
+
+
 @skipif_yask
-@pytest.mark.parametrize('SymbolType, dimension', [
+@pytest.mark.parametrize('SymbolType, dim', [
     (Function, x), (Function, y),
     (TimeFunction, x), (TimeFunction, y), (TimeFunction, t),
 ])
-def test_stencil_derivative(grid, shape, SymbolType, dimension):
+def test_stencil_derivative(grid, shape, SymbolType, dim):
     """Test symbolic behaviour when expanding stencil derivatives"""
+    i = dim(grid)  # Known pytest issue: cannot promptly use fixtures with parametrize
```

Comment:
What is the issue? Maybe leave a link in comments?

Reply:
the comment already explains it: you can't use fixtures within ``parametrize``

Reply:
A link to the issue tracker on pytest's website would still be useful.

Reply:
ok